oops/exceptions/excep3.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+using namespace std;
+
+// for creating our own exception
+class InvalidAmountError : public runtime_error
+{
+public:
+    InvalidAmountError(const string &msg) : runtime_error(msg) {};
+};
+
+class BankAccount
+{
+    string name;
+    int balance, acc_no;
+
+public:
+    BankAccount(string n, int b, int ac_no)
+    {
+        this->name = n;
+        this->balance = b;
+        this->acc_no = ac_no;
+    }
+
+    void deposit(int amount)
+    {
+        if (amount < 0)
+        {
+            throw InvalidAmountError("invalid deposit");
+        }
+        balance += amount;
+        cout << "after deposit : " << balance << endl;
+    }
+
+    void withdraw(int amount)
+    {
+        if (amount > 0 && amount <= balance)
+        {
+            balance -= amount;
+            cout << "after withdraw : " << balance << endl;
+        }
+        else
+        {
+        }
+        throw "insufficient balance";
+    }
+};
```

Comment:
```suggestion
}
```
This should be throwing a compilation error

Reply:
no it is this is used after the end of the class